File: forum-backend/src/main/java/com/projects/pes/forumbackend/pojo/Forum.java
package com.projects.pes.forumbackend.pojo;

import com.projects.pes.forumbackend.entities.FacultyEntity;
import com.projects.pes.forumbackend.entities.PostEntity;
import com.projects.pes.forumbackend.entities.ResourceEntity;
import com.projects.pes.forumbackend.entities.UserEntity;
import jakarta.persistence.*;

import java.util.Set;
import java.util.UUID;

public record Forum(
         UUID id,
         Faculty admin,
         String name,
         Set<Post> posts,
         Set<Resource> resources,
         Set<UserInAForum> users
) {
}

File: student-forums-frontend/src/app/forum.ts
import { Faculty } from "./faculty";
import { Post } from "./post";
import { User } from "./user";

export class Forum {
    id:string;
    admin: Faculty|null;
    name: string;
    posts: Array<Post>;
    resources: null;
    users: Array<User>;
    constructor(id:string, name: string) {
        this.id = id;
        this.name = name;
        this.admin = null;
        this.resources = null;
        this.users = [];
        this.posts = []
    }
}
